### ISSUE TITLE
Add "Sign Up" method

### ISSUE THREAD

**Issue by erichaus** [fabscale/ember-cognito-identity#721]
Hello, I'm investigating this as a replacement for [ember-cognito](https://github.com/adopted-ember-addons/ember-cognito) due to a blocking bug brought about by reliance on [@aws-amplify](https://github.com/facebook/react-native/issues/33540).

The only thing stopping me is that this library is missing a `signup` method. Is there a reason for that? Can we add it?

If we can add that I think this is a great replacement for `ember-cognito`.

At first I was going to suggest making this compatible with [ember-simple-auth](https://github.com/simplabs/ember-simple-auth), but I'm not sure it's even necessary and I like the idea of thin packages.

Thanks!

**Comment by hoIIer** [comment]
`cognito._userpool.signUp()` exists so adding this should be trivial?

**Comment by hoIIer** [comment]
Was able to get it working via below:
```
import AmazonCognitoIdentity from 'amazon-cognito-identity-js';

        ...

        const attributes = [
          new AmazonCognitoIdentity.CognitoUserAttribute({
            Name: 'email',
            Value: email,
          }),
          new AmazonCognitoIdentity.CognitoUserAttribute({
            Name: 'custom:initial_group_id',
            Value: 'Manufacturers',
          }),
          new AmazonCognitoIdentity.CognitoUserAttribute({
            Name: 'custom:signup_code',
            Value: code,
          }),
          new AmazonCognitoIdentity.CognitoUserAttribute({
            Name: 'custom:foo',
            Value: 'bar',
          }),
        ];

        await this.cognito.userPool.signUp(
          email,
          password,
          attributes,
        );
```

**Comment by mydea** [comment]
Hi, no specific reason for not adding this except that I didn't need it so far ;)
I'd be happy to accept a PR adding this functionality. Other than this, it should also be totally fine to use the user pool like you did in your example!

**Comment by hoIIer** [comment]
Hey, ok yeah the signUp works needs better handling for promises like below:
https://github.com/aws-amplify/amplify-js/blob/main/packages/auth/src/Auth.ts#L362-L388

Will see if I can take a crack at it later this week perhaps

**Comment by mydea** [comment]
Yeah, you'll need to wrap it like the other methods - I guess it should be realtively straightforward to follow the same schema as for all the other stuff, but let me know if you need guidance somewhere!
